File: RecommendationEngine/Models/Application/Asset/Asset.cs
namespace Models.Application.Asset
{
    public abstract class Asset
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string DisplayText { get; set; }
        public string EnergyType { get; set; }
        public string TimeZone { get; set; }
        public string ElementPath { get; set; }
        public string AssetType { get; set; }
        public double AcPower { get; set; }
    }
}

File: RecommendationEngine/RecommendationEngine/FrontEnd/src/api/models/Asset.ts
//You have to make sure that the method that fetches this object
//have exactly this attribute, or it will break and it will be hard to debug.
export interface Asset {
    id: number,
    name: string,
    displayText: string,
    energyType: string,
    timeZone: string,
    elementPath: string,
    acPower: string,
    children: Asset[],
    assetType: string
}

export interface MultiSelectTreeViewAsset {
    key: number,
    title: string,
    value: Asset,
    children: MultiSelectTreeViewAsset[],
}